google/cloud/spanner_v1/database.py

```diff
@@ -887,8 +906,16 @@ def run_in_transaction(self, func, *args, **kw):
         # Check out a session and run the function in a transaction; once
         # done, flip the sanity check bit back.
         try:
-            with SessionCheckout(self._pool) as session:
-                return session.run_in_transaction(func, *args, **kw)
+            observability_options = getattr(self, "observability_options", {})
+            if isinstance(observability_options, dict):
+                observability_options["db_name"] = self.name
```

Comment:
same here, we can add this at a single place in observability_options getter property?

Reply:
Done, thanks!